xpsi/Instrument.py

```diff
@@ -213,6 +213,14 @@ def energy_edges(self):
     def energy_edges(self, energy_edges):
         """ Set the energy edges in keV. """
 
+        try:
+            assert energy_edges.ndim == 1
+            assert (energy_edges >= 0.0).all()
+            assert energy_edges.shape[0] == self._matrix.shape[1] + 1
+            assert not (energy_edges[1:] <= energy_edges[:-1]).any()
+        except AssertionError:
+            raise EdgesError('Energy edges must be in a one-dimensional '
+                             'array, and must be postive and increasing.')
         if not isinstance(energy_edges, _np.ndarray):
```

Comment:
you should check that it's an ndarray first, before trying to lookup attributes like `ndim`.